server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -827,6 +844,27 @@ public Query rangeQuery(
                     (dv, n) -> new DelegateDocValuesField(new Longs(new LongsSupplier(dv)), n)
                 );
             }
+
+            @Override
+            public QueryableExpression asQueryableExpression(String field, boolean hasDocValues, SearchExecutionContext context) {
+                return LongQueryableExpression.field(field, new LongQueryableExpression.IntQueries() {
```

Comment:
`integer` fields are exposed to scripts as `long`s so we have to use `LongQueryableExpression` to do arithmetic with `long`. But we can use the query functionality of integers.